src/ValueObjects/Transporter/Payload.php

```diff
@@ -164,15 +164,23 @@ public function toRequest(BaseUri $baseUri, Headers $headers, QueryParams $query
             if ($this->contentType === ContentType::MULTIPART) {
                 $streamBuilder = new MultipartStreamBuilder($psr17Factory);
 
-                /** @var array<string, StreamInterface|string|int|float|bool> $parameters */
+                /** @var array<string, StreamInterface|string|int|float|bool|array<int, string>> $parameters */
                 $parameters = $this->parameters;
 
                 foreach ($parameters as $key => $value) {
                     if (is_int($value) || is_float($value) || is_bool($value)) {
                         $value = (string) $value;
                     }
 
-                    $streamBuilder->addResource($key, $value);
+                    // Support only one level of nested arrays
+                    if (is_array($value)) {
+                        foreach ($value as $nestedKey => $nestedValue) {
+                            $streamBuilder->addResource($key.'['.$nestedKey.']', $nestedValue);
```

Comment:
This didn't work for me using the provided syntax: `'timestamp_granularities' => ['word'],`

For this to work we need to do something like
```suggestion
                            $finalkey = is_int($nestedKey) ? $key.'[]' : $key.'['.$nestedKey.']';
                            $streamBuilder->addResource($finalkey, $nestedValue);
```